src/ethereum_spec_tools/lint.py

```diff
@@ -151,20 +177,25 @@ def lint(
         """
         Walks the sources for each hardfork and emits Diagnostic messages.
         """
-        if position == 0:
-            # Nothing to compare against!
-            return []
-
-        all_previous = dict(walk_sources(forks[position - 1]))
         all_current = dict(walk_sources(forks[position]))
-
-        items = (
-            (k, v, all_previous.get(k, None)) for (k, v) in all_current.items()
-        )
-
         diagnostics: List[Diagnostic] = []
-        for (name, current, previous) in items:
-            diagnostics += self.compare(name, current, previous)
+        if position == 0:
+            for (name, source) in all_current.items():
+                # No need to run compare since this is the first fork
+                diagnostics += self.check_import(forks, position, name, source)
```

Comment:
I think this feature might make more sense as a separate `Lint` subclass with its own visitor?

Reply:
Have implemented the `ImportHygiene` sub-class of `Lint` with its own visitor `ImportHygieneVisitor`.
Restored `PatchHygiene` and `PatchHygieneVisitor` to their original form.

The `lint` module was starting to get unwieldy. So I just separated all the visitors out into a different module.